src/monodroid/jni/android-system.cc

```diff
@@ -252,30 +262,30 @@ AndroidSystem::monodroid_get_system_property (const char *name, char **value)
 	}
 
 	if (len >= 0 && value) {
-		*value = new char [len+1];
+		*value = new char [static_cast<size_t>(len) + 1];
```

Comment:
I wonder if we should introduce a new `CHECK_SIZE()` macro which uses `__builtin_add_overflow()`?

Reply:
hmm, maybe an inline function - I'd rather avoid macros if possible, but it's a good idea